analysis/warriorarms/src/modules/features/SpellUsable.js

```diff
@@ -1,21 +1,36 @@
 import SPELLS from 'common/SPELLS';
+import { EventType } from 'parser/core/Events';
+import Abilities from 'parser/core/modules/Abilities';
 import CoreSpellUsable from 'parser/shared/modules/SpellUsable';
 
 class SpellUsable extends CoreSpellUsable {
   static dependencies = {
     ...CoreSpellUsable.dependencies,
+    abilities: Abilities,
   };
-
-  beginCooldown(spellId, ...args) {
+  onEvent(event) {
+    super.onEvent(event);
     // Tactician passive: You have a 1.40% chance per Rage spent on damaging abilities to reset the remaining cooldown on Overpower.
-    if (spellId === SPELLS.OVERPOWER.id) {
-      if (this.isOnCooldown(spellId)) {
-        this.endCooldown(spellId);
+    // normally charges dont count down simultaneously. these do
+    if (event.type === EventType.ApplyBuff || event.type === EventType.RefreshBuff) {
+      if (event.ability.guid === SPELLS.TACTICIAN.id) {
+        if (this.isOnCooldown(SPELLS.OVERPOWER.id)) {
```

Comment:
instead of using nested ifs like this, use `&&` on the condition (suggestion doesn't handle multiple lines but simple idea)

```suggestion
        if ((event.type === EventType.ApplyBuff || event.type === EventType.RefreshBuff) && event.ability.guid === SPELLS.TACTICIAN.id && this.isOnCooldown(SPELLS.OVERPOWER.id)) {
```